src/oop/static_modifier/medium/q6/MainTest.java

```diff
@@ -0,0 +1,83 @@
+package oop.static_modifier.medium.q6;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    ObjectTest testClass;
+    String className = "TextProcessing";
+    String methodName1 = "canadianize";
+    String methodName2 = "extractVowels";
+
+    public String f1(String s) {
+        return s + ", eh?";
+    }
+
+    public String f2(String s) {
+        String str = "";
+        for (int i = 0; i < s.length(); i++) {
+            char ch = s.charAt(i);
+            if (ch != 'a' && ch != 'e' && ch != 'i' && ch != 'o' && ch != 'u')
+                str += ch;
+        }
+        return str;
+    }
+
+    @BeforeEach
+    public void setup() throws Throwable {
+        String classString = "oop.static_modifier.medium.q6." + className;
+        testClass = new ObjectTest(classString);
+    }
+
+    private static Stream<String> inputProvider() {
+        return Stream.of(
+                "Black", "Blue", "Red", "Brown", "Hello there", "It is hot", "How is the weather", "How's your day?"
```

Comment:
you could split this into two input providers so that you could hardcode the answers instead of using the same methods as the answer